frontend/packages/dev-console/src/utils/__tests__/add-resources-menu-utils.spec.tsx

```diff
@@ -0,0 +1,138 @@
+    const { resource } = getTopologyData(MockResources, ['deployments']);
+    const url = new URL(
+      getAddPageUrl(resource, ImportOptions.CONTAINER, true),
+      'https://mock.test.com',
+    );
+    // Check that url contains all expected params
+    expect(url.pathname).toBe('/deploy-image/ns/testproject1');
+    expect(url.searchParams.get('preselected-ns')).toBe('testproject1');
+    expect(url.searchParams.get('application')).toBe('application-1');
+    expect(url.searchParams.get('isKnativeDisabled')).toBe('true');
+    expect(Array.from(url.searchParams.entries())).toHaveLength(3);
+  });
+
+  it('should return the page url with proper queryparams for catalog flow', () => {
+    const { resource } = getTopologyData(MockResources, ['deployments']);
+    const url = new URL(
+      getAddPageUrl(resource, ImportOptions.CATALOG, true),
+      'https://mock.test.com',
+    );
+    // Check that url contains all expected params
+    expect(url.pathname).toBe('/catalog');
+    expect(url.searchParams.get('application')).toBe('application-1');
+    expect(url.searchParams.get('isKnativeDisabled')).toBe('true');
+    expect(Array.from(url.searchParams.entries())).toHaveLength(2);
+  });
+
+  it('should return the page url with proper queryparams for dockerfile flow', () => {
+    const { resource } = getTopologyData(MockResources, ['deployments']);
+    const url = new URL(
+      getAddPageUrl(resource, ImportOptions.DOCKERFILE, true),
+      'https://mock.test.com',
+    );
+    // Check that url contains all expected params
+    expect(url.pathname).toBe('/import/ns/testproject1');
+    expect(url.searchParams.get('importType')).toBe('docker');
+    expect(url.searchParams.get('application')).toBe('application-1');
+    expect(url.searchParams.get('isKnativeDisabled')).toBe('true');
+    expect(Array.from(url.searchParams.entries())).toHaveLength(3);
+  });
+
+  it('should return the page url with proper queryparams for database flow', () => {
+    const { resource } = getTopologyData(MockResources, ['deployments']);
+    const url = new URL(
+      getAddPageUrl(resource, ImportOptions.DATABASE, true),
+      'https://mock.test.com',
+    );
+    // Check that url contains all expected params
+    expect(url.pathname).toBe('/catalog/ns/testproject1');
+    expect(url.searchParams.get('category')).toBe('databases');
+    expect(url.searchParams.get('application')).toBe('application-1');
+    expect(url.searchParams.get('isKnativeDisabled')).toBe('true');
+    expect(Array.from(url.searchParams.entries())).toHaveLength(3);
+  });
+
+  it('it should return a valid kebabAction on invoking createKebabAction', () => {
+    const { resource } = getTopologyData(MockResources, ['deployments']);
+    const kebabAction = createKebabAction('From Git', <GitAltIcon />, ImportOptions.GIT);
+    expect(kebabAction(resource, true)).toHaveProperties([
+      'label',
+      'icon',
+      'path',
+      'href',
+      'accessReview',
+    ]);
+  });
```

Comment:
You use `toHaveProperties` once here, and I don't think you're doing anything special here that warranted hiding the logic under the utility you created. Recommend a more direct approach, it's just an object and you know what structure you're getting (a `KebabOption`) so test for that:

```
  it('should return a valid KebabOption on invoking a new KebabAction', () => {
    const { resource } = getTopologyData(MockResources, ['deployments']);
    const icon = <GitAltIcon />;
    const hasApplication = true;
    const label = 'From Git';

    const kebabAction: KebabAction = createKebabAction(label, icon, ImportOptions.GIT);
    const kebabOption: KebabOption = kebabAction(resource, hasApplication);

    expect(kebabOption.label).toEqual(label);
    expect(kebabOption.icon).toEqual(icon);
    expect(kebabOption.path).toEqual('Add to Application');
    expect(kebabOption.href).toEqual(getAddPageUrl(resource, ImportOptions.GIT, hasApplication));
    expect(kebabOption.accessReview).toEqual(asAccessReview(DeploymentModel, resource, 'create'));
  });
```
Should be noted I changed your test name as well, you had a couple issues with it.

Also delete the `toHaveProperties` extension at the top of the file.

I used functions calls for `href` and `accessReview` as you're not really testing an explicit output, you're just using what they provide.

Reply:
fixed it